src/AndersonAcceleration.f90

```diff
@@ -72,183 +71,288 @@ MODULE AndersonAccelerationTypes
-
+CHARACTER(LEN=*),PARAMETER :: modName='AndersonAccelerationTypes'
 !
 !===============================================================================
 CONTAINS
 !
 !-------------------------------------------------------------------------------
-!> @brief Initializes the Anderson Acceleration Type with a parameter list
+!> @brief Initializes the Anderson Acceleration Type with an input parameter list
 !> @param pList the parameter list
+!> @param solver The Anderson Acceleration solver to act on
+!> @param Params The Anderson options parameter list
+!> @param ce The computing environment to use for the calculation
 !>
-!> @param solver The anderson acceleration solver to act on
-!> @param MPIEnv The MPI environment description
-!> @param Params The aprameterlist
-!>
-!> This routine initializes the data spaces for the NOX Anderson Acceleration
-!>
-SUBROUTINE init_AndersonAccelerationType(solver,MPIEnv,Params)
+SUBROUTINE init_AndersonAccelerationType(solver,ce,Params)
   CHARACTER(LEN=*),PARAMETER :: myName='init_AndersonAccelerationType'
   CLASS(AndersonAccelerationType),INTENT(INOUT) :: solver
-  TYPE(MPI_EnvType),INTENT(IN),TARGET :: MPIEnv
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
   TYPE(ParamType),INTENT(IN) :: Params
 
-  TYPE(ParamType) :: validParams
-  INTEGER(SIK) :: n,nlocal,depth,start
-  REAL(SRK) :: beta
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-  TYPE(ParamType) :: tmpPL
-#endif
-
-  !Check to set up required and optional param lists.
-  !IF(.NOT.AndersonType_Paramsflag) CALL AndersionType_Declare_ValidParams()
-  !Validate against the reqParams and OptParams
-  validParams=Params
-  !CALL validParams%validate(AndersonType_reqParams)
+  TYPE(ParamType) :: LSparams
+  INTEGER(SIK) :: i,j,m
 
-  IF(.NOT. MPIEnv%isInit()) THEN
-    CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-        modName//'::'//myName//' - MPI Environment is not initialized!')
-  ELSE
-    solver%MPIparallelEnv => MPIEnv
-  ENDIF
+  REQUIRE(Params%has('AndersonAccelerationType->n'))
+  REQUIRE(Params%has('AndersonAccelerationType->depth'))
+  REQUIRE(Params%has('AndersonAccelerationType->beta'))
+  REQUIRE(Params%has('AndersonAccelerationType->start'))
 
-  n=0
-  nlocal=0
-  depth=-1
-  beta=0.0_SRK
-  start=1
   !Pull Data from Parameter List
-  CALL validParams%get('AndersonAccelerationType->n',n)
-  CALL validParams%get('AndersonAccelerationType->nlocal',nlocal)
-  CALL validParams%get('AndersonAccelerationType->depth',depth)
-  CALL validParams%get('AndersonAccelerationType->beta',beta)
-  CALL validParams%get('AndersonAccelerationType->start',start)
-  CALL validParams%clear()
+  CALL Params%get('AndersonAccelerationType->n',solver%n)
+  CALL Params%get('AndersonAccelerationType->depth',solver%depth)
+  CALL Params%get('AndersonAccelerationType->beta',solver%beta)
+  CALL Params%get('AndersonAccelerationType->start',solver%start)
 
   IF(.NOT. solver%isInit) THEN
```

Comment:
So, instead of a giant IF block, just `REQUIRE(.NOT.solver%isInit)` instead

Reply:
I like it.